storm-core/src/jvm/backtype/storm/utils/ShellProcess.java

```diff
@@ -48,8 +47,26 @@ public ShellProcess(String[] command) {
         this.command = command;
     }
 
+    public void setEnv(Map<String, String> env) {
+        this.env = env;
+    }
+
+    private void modifyEnvironment(Map<String, String> buildEnv) {
+        for (Map.Entry<String, String> entry : env.entrySet()) {
+            if (entry.getKey().equals("PATH")) {
```

Comment:
Just on the safe side: Let's use the form `"PATH".equals ...` in case the key is null for whatever reason.

I think with a Map this is impossible, but should there be a code change later, we can trivially avoid an NPE here.